src/stp/new_plays/GetBallRisky.cpp

```diff
@@ -0,0 +1,109 @@
+#include "stp/invariants/game_states/NormalPlayGameStateInvariant.h"
+#include "stp/new_roles/BallGetter.h"
+#include "stp/new_roles/Keeper.h"
+
+namespace rtt::ai::stp::play {
+
+GetBallRisky::GetBallRisky() : Play() {
+    startPlayInvariants.clear();
+    startPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
+
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
+
+    roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{std::make_unique<role::Keeper>(role::Keeper("keeper")),
+                                                                                 std::make_unique<role::BallGetter>(role::BallGetter("ball_getter")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_0")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_1")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_2")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_0")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_1")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_2")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_0")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_1")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_2"))};
+}
+
+uint8_t GetBallRisky::score(world_new::World* world) noexcept { return 120; }
+
+void GetBallRisky::calculateInfoForRoles() noexcept {
+    auto enemyRobots = world->getWorld()->getThem();
+    auto enemyAttacker = world->getWorld()->getRobotClosestToBall(world_new::them);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(), [&](const auto enemyRobot) -> bool { return enemyRobot->getId() == enemyAttacker->getId(); }));
```

Comment:
This is a copy of getThem(), right? Otherwise we are killing robots here... :smile:

Reply:
it is a const reference to the non-owning container of RobotViews of the enemy team, and the world where we get that from should be updated every tick.